storage/src/main/java/zipkin2/storage/kafka/KafkaAutocompleteTags.java

```diff
@@ -34,62 +42,105 @@
  * for scatter gather data from different instances.
  */
 public class KafkaAutocompleteTags implements AutocompleteTags {
+  static final String HTTP_BASE_URL = "http://%s:%d";
+
   final KafkaStreams traceStoreStream;
 
   KafkaAutocompleteTags(KafkaStorage storage) {
     traceStoreStream = storage.getTraceStoreStream();
   }
 
   @Override public Call<List<String>> getKeys() {
-    ReadOnlyKeyValueStore<String, Set<String>> autocompleteTagsStore =
-        traceStoreStream.store(AUTOCOMPLETE_TAGS_STORE_NAME,
-            QueryableStoreTypes.keyValueStore());
-    return new GetKeysCall(autocompleteTagsStore);
+    return new GetTagKeysCall(traceStoreStream);
   }
 
   @Override public Call<List<String>> getValues(String key) {
-    ReadOnlyKeyValueStore<String, Set<String>> autocompleteTagsStore =
-        traceStoreStream.store(AUTOCOMPLETE_TAGS_STORE_NAME,
-            QueryableStoreTypes.keyValueStore());
-    return new GetValuesCall(autocompleteTagsStore, key);
+    return new GetTagValuesCall(traceStoreStream, key);
   }
 
-  static class GetKeysCall extends KafkaStreamsStoreCall<List<String>> {
-    final ReadOnlyKeyValueStore<String, Set<String>> autocompleteTagsStore;
+  static class GetTagKeysCall extends Call.Base<List<String>> {
+    static final Gson GSON = new Gson();
+
+    final KafkaStreams traceStoreStream;
 
-    GetKeysCall(ReadOnlyKeyValueStore<String, Set<String>> autocompleteTagsStore) {
-      this.autocompleteTagsStore = autocompleteTagsStore;
+    GetTagKeysCall(KafkaStreams traceStoreStream) {
+      this.traceStoreStream = traceStoreStream;
     }
 
-    @Override protected List<String> query() {
-      List<String> keys = new ArrayList<>();
-      autocompleteTagsStore.all().forEachRemaining(keyValue -> keys.add(keyValue.key));
-      return keys;
+    @Override protected List<String> doExecute() throws IOException {
+      return traceStoreStream.allMetadataForStore(AUTOCOMPLETE_TAGS_STORE_NAME)
+          .parallelStream()
+          .map(KafkaAutocompleteTags::httpClient)
+          .map(httpClient -> {
+            AggregatedHttpResponse response = httpClient.get("/autocomplete-tags")
+                .aggregate()
+                .join();
+            if (!HttpStatus.OK.equals(response.status())) return null;
+            return response.contentUtf8();
+          })
+          .map(content -> {
+            Set<String> set = GSON.fromJson(content, Set.class);
+            return set;
+          })
+          .flatMap(Collection::stream)
+          .distinct()
+          .collect(Collectors.toList());
+    }
+
+    @Override protected void doEnqueue(Callback<List<String>> callback) {
+      try {
+        callback.onSuccess(doExecute());
```

Comment:
@adriancole Here's what you've schooled me about :) Do you think we should try to make new code that implements `doEnqueue` not block or just consider the ship sailed?

Reply:
I think probably the best idea is to re-use the http call code so that lessons schooling etc are contained. I didn't do a full review here and doing so would indeed repeat everything covered in the ES mutually schooled!

Reply:
one way to temporarily solve this is to add a dependency on storage-elasticsearch and disclose that drift might happen.. or shade zipkin2.elasticsearch.internal.client